test/kubernetes_unittest.cc

```diff
@@ -72,27 +77,296 @@ class KubernetesTest : public ::testing::Test {
       throw(json::Exception) {
     return reader->UpdateServiceToPodsCache(endpoints, is_deleted);
   }
+
+  void SetUp() override {
+    config = CreateConfig();
+    store.reset(new MetadataStore(*config));
+    reader.reset(new KubernetesReader(*config,
+                                      *store,
+                                      nullptr));  // Don't need HealthChecker.
+  }
+
+  virtual std::unique_ptr<Configuration> CreateConfig() = 0;
+
+  std::unique_ptr<Configuration> config;
+  std::unique_ptr<MetadataStore> store;
+  std::unique_ptr<KubernetesReader> reader;
 };
 
-TEST_F(KubernetesTest, GetNodeMetadata) {
-  Configuration config(std::istringstream(
-    "KubernetesClusterName: TestClusterName\n"
-    "KubernetesClusterLocation: TestClusterLocation\n"
-    "MetadataIngestionRawContentVersion: TestVersion\n"
-    "InstanceResourceType: gce_instance\n"
-    "InstanceZone: TestZone\n"
-    "InstanceId: TestID\n"
-  ));
-  Environment environment(config);
-  KubernetesReader reader(config, nullptr);  // Don't need HealthChecker.
+class KubernetesTestNoInstance : public KubernetesTest {
+ protected:
+  std::unique_ptr<Configuration> CreateConfig() override {
+    return std::unique_ptr<Configuration>(
+      new Configuration(std::istringstream(
+        "KubernetesClusterName: TestClusterName\n"
+        "KubernetesClusterLocation: TestClusterLocation\n"
+        "MetadataIngestionRawContentVersion: TestVersion\n"
+      )));
+  }
+};
+
+TEST_F(KubernetesTestNoInstance, GetNodeMetadata) {
   json::value node = json::object({
     {"metadata", json::object({
       {"name", json::string("testname")},
       {"creationTimestamp", json::string("2018-03-03T01:23:45.678901234Z")},
     })}
   });
   const auto m =
-      GetNodeMetadata(reader, node->As<json::Object>(), Timestamp(), false);
+      GetNodeMetadata(*reader, node->As<json::Object>(), Timestamp(), false);
+  EXPECT_EQ(1, m.ids().size());
```

Comment:
nit: indentation seems off

Reply:
Where? This lines up with the above `const auto`...